Add notification example using Telegram (#585)
vmware-samples/vcenter-event-broker-appliance: examples/knative/powercli/kn-pcli-telegram/README.md

# kn-pcli-telegram

> Example Knative PowerCLI function for sending a [Telegram](https://telegram.org/) message when a VM is migrated and also resides in a specific vSphere Resource Pool.

Reviewer: I understand the script will also work if VM is not part of RP right?

If so, suggesting to reword:

```suggestion
Example Knative PowerCLI function for sending a [Telegram](https://telegram.org/) message when a VM is migrated and optionally filter by its location in a vSphere Resource Pool.
```

![](screenshots/screenshots-01.png)

# Step 1 - Build

Create the container image locally to test your function logic.

```
export TAG=<version>
docker build -t <docker-username>/kn-pcli-telegram:${TAG} .
```

# Step 2 - Test

Verify the container image works by executing it locally.

Change into the `test` directory
```console
cd test
```

Update the following variable names within the `docker-test-env-variable` file

* TELEGRAM_BOT_API_KEY - Telegram Bot API key
* TELEGRAM_GROUP_CHAT_ID - Telegram Group Chat ID

> * VCENTER_RESOURCE_POOL_FILTER - Name of vSphere Resource Pool to filter for and only sent Telegram if migrated VM is located in RP

Reviewer: ```suggestion
* VCENTER_RESOURCE_POOL_FILTER - Name of vSphere Resource Pool to filter for and only send Telegram if migrated VM is located in RP
```

* VCENTER_SERVER - IP Address or FQDN of the vCenter Server to connect to check for parent vSphere Resource Pool
* VCENTER_USERNAME - vCenter account with permission to check for parent vSphere Resource Pool
* VCENTER_PASSWORD - vCenter credentials to account with permission to check for parent vSphere Resource Pool
* VCENTER_CERTIFCATE_ACTION - Set-PowerCLIConfiguration Action to configure when connection fails due to certificate error, default is Fail. (Possible values: Fail, Ignore or Warn)

Start the container image by running the following command:

```console
docker run -e FUNCTION_DEBUG=true -e PORT=8080 --env-file docker-test-env-variable -it --rm -p 8080:8080 <docker-username>/kn-pcli-telegram:${TAG}
```

In a separate terminal, run either `send-cloudevent-test.ps1` (PowerShell Script) or `send-cloudevent-test.sh` (Bash Script) to simulate a CloudEvent payload being sent to the local container image

```console
Testing Function ...
See docker container console for output

# Output from docker container console

> 08/11/2021 14:45:44 - PowerShell HTTP server start listening on 'http://*:8080/'

Reviewer: I guess these logs are from a <1.3 base image tag?

Author: Correct
